ui/src/common/QualityInfo.test.js

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react'
+import { cleanup, render } from '@testing-library/react'
+import { QualityInfo } from './QualityInfo'
+
+describe('<QualityInfo />', () => {
```

Comment:
We are missing tests for other conditions: 
- A lossy format (should add the bitrate)
- Invalid data (ex: how the component behaves if we don't pass a `record`)

Reply:
How to add test for if we don't pass a ``` record ```? Because if we don't pass a record it will not get rendered so how will we grab that?

Reply:
You just have to make sure that it does not throw an exception when it is `undefined`.

Reply:
You just have to make sure that it does not throw an exception when it is undefined.

Reply:
But when i will pass info as record prop as an empty object how will it read ```record.suffix or record.bitRate``` . (I can write ```.toBeNull``` but deconstruction in QualityInfo happes before it)

Reply:
`null` or `undefined` != `{}`

If you use `defaultProps` and set `record` to be `{}` as default, the code should not break in a situation where the `record` is not present (thus `undefined`. That's what you should write a test for, to make sure this situation is covered.

Reply:
I tried it a lot (I dont want to ask it again and again sir) but we have ``` suffix = suffix.toUpperCase()``` too.

Reply:
Yes, I know. Then you need to fix the code to not break in this situation. Maybe return a Chip with `label="???"` in this case?

Reply:
What is the need to write tests when we use ``` {data && <QualityInfo record={data} sortable={false} />}``` in PlayerToolbar.js. Its better to leave the test for it rather than tweaking the code in QualityInfo. IMO. But if you want a test i will tweak it.

Reply:
We don't always use `{data && <QualityInfo/>}`: 
https://github.com/navidrome/navidrome/blob/f9db8c3c49b0723fec364312ab6a84b9c0e648c6/ui/src/album/AlbumSongs.js#L123

You may say that in the case I linked here, it is responsibility of React Admin to always pass the `record` and while this is true, I've seen a lot of instances where it does not pass it (probably caused by some bug) and breaks the UI with an exception. That's why I try to always check for such cases.

Also, what if we want to use the component in another place and forget to add the check? Adding this check in the component itself is best practice and even has a name: it is called "defensive programming".